$OpenBSD: patch-mysys_crc32_crc32c_cc,v 1.1 2021/01/09 21:03:10 jasper Exp $

Extend ifdefs so we can define arch_ppc_crc32 which in turns fixes
an undefined reference to my_checksum().

Index: mysys/crc32/crc32c.cc
--- mysys/crc32/crc32c.cc.orig
+++ mysys/crc32/crc32c.cc
@@ -465,12 +465,16 @@ uint32_t ExtendPPCImpl(uint32_t crc, const char *buf, 
   return crc32c_ppc(crc, (const unsigned char *)buf, size);
 }
 
-#if __linux__
+#if defined(__linux__) || defined(__OpenBSD__)
 static int arch_ppc_probe(void) {
   arch_ppc_crc32 = 0;
 
 #if defined(__powerpc64__)
+  #if __linux__
   if (getauxval(AT_HWCAP2) & PPC_FEATURE2_VEC_CRYPTO) arch_ppc_crc32 = 1;
+  #elif __OpenBSD__
+  arch_ppc_crc32 = 1;
+  #endif
 #endif /* __powerpc64__ */
 
   return arch_ppc_crc32;
